libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -619,38 +619,39 @@ bool AP_AHRS_NavEKF::get_quaternion(Quaternion &quat) const
 // return secondary attitude solution if available, as eulers in radians
 bool AP_AHRS_NavEKF::get_secondary_attitude(Vector3f &eulers) const
 {
-    switch (active_EKF_type()) {
+    EKFType secondary_ekf_type;
+    if (!get_secondary_EKF_type(secondary_ekf_type)) {
+        return false;
+    }
+
+    switch (secondary_ekf_type) {
+
     case EKFType::NONE:
-        // EKF is secondary
-        switch ((EKFType)_ekf_type.get()) {
-#if HAL_NAVEKF3_AVAILABLE
-        case EKFType::THREE:
-            EKF3.getEulerAngles(-1, eulers);
-            return _ekf3_started;
-#endif
-#if HAL_NAVEKF2_AVAILABLE
-        case EKFType::TWO:
-            EKF2.getEulerAngles(-1, eulers);
-            return _ekf2_started;
-#endif
-        default:
-            return false;
-        }
+        // DCM is secondary
+        eulers = _dcm_attitude;
+        return true;
+
 #if HAL_NAVEKF2_AVAILABLE
     case EKFType::TWO:
+        // EKF2 is secondary
+        EKF2.getEulerAngles(-1, eulers);
+        return _ekf2_started;
 #endif
 
 #if HAL_NAVEKF3_AVAILABLE
     case EKFType::THREE:
+        // EKF3 is secondary
+        EKF3.getEulerAngles(-1, eulers);
+        return _ekf3_started;
 #endif
 
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     case EKFType::SITL:
+        // SITL is secondary (should never happen)
 #endif
-        // DCM is secondary
-        eulers = _dcm_attitude;
-        return true;
+        return false;
```

Comment:
this one should be in the SITL #if as it serve no one now, unless we want a default case

Reply:
great, thanks for the review, fixed!